packages/react/src/types/temporal-adapter.ts

```diff
@@ -0,0 +1,319 @@
+import { TemporalTimezone, TemporalSupportedObject, TemporalSupportedValue } from './temporal';
+
+export interface TemporalAdapterFormats {
+  /**
+   * The 4-digit year.
+   * @example "2019"
+   */
+  yearPadded: string;
+  /**
+   * The month with leading zeros.
+   * @example "08"
+   */
+  monthPadded: string;
+  /**
+   * The day of the month with leading zeros.
+   * @example "01"
+   */
+  dayOfMonthPadded: string;
+  /**
+   * The hours with leading zeros, 24-hour clock.
+   * @example "01", "23"
+   */
+  hours24hPadded: string;
+  /**
+   * The hours with leading zeros, 12-hour clock.
+   * @example "01", "11"
+   */
+  hours12hPadded: string;
+  /**
+   * The minutes with leading zeros.
+   * @example "01", "59"
+   */
+  minutesPadded: string;
+  /**
+   * The seconds with leading zeros.
+   * @example "01", "59"
+   */
+  secondsPadded: string;
+  /**
+   * The day of the month without leading zeros.
+   * @example "1"
+   */
+  dayOfMonth: string;
+  /**
+   * The name of the day of the week.
+   * @example "Wednesday"
+   */
+  weekday: string;
+  /**
+   * The abbreviated name of the day of the week.
+   * @example "Wed"
+   * */
+  weekday3Letters: string;
+  /**
+   * The meridiem.
+   * @example "AM"
+   */
+  meridiem: string;
+  /**
+   * The localized date format including year, month, day and weekday.
+   * @example "Wednesday, August 6, 2014"
+   */
+  localizedDateWithFullMonthAndWeekDay: string;
+  /**
+   * The non-localized month and year format.
+   * @example "August 2014"
+   */
+  fullMonthAndYear: string;
+}
+
+export type DateBuilderReturnType<T extends string | null> = [T] extends [null]
+  ? null
+  : TemporalSupportedObject;
+
+export interface TemporalAdapter {
+  isTimezoneCompatible: boolean;
```

Comment:
This would be redundant if we would never add an adapter that does not support timezones (i.e. `AdapterMomentHijri` or `AdapterMomentJalaali` from mui-x: https://github.com/search?q=repo%3Amui%2Fmui-x+%22isTimezoneCompatible+%3D+false%22&type=code)

Reply:
I think it's safer to keep it for now (doesn't cost us anything) but yeah we should be able to drop it eventually